stdlib/src/memory/anypointer.mojo

```diff
@@ -42,15 +42,13 @@ struct AnyPointer[T: Movable](
     """The underlying pointer."""
 
     @always_inline
-    fn __init__() -> Self:
+    fn __init__(inout self):
         """Create a null pointer.
 
         Returns:
             A null pointer.
         """
-        return Self {
-            value: __mlir_attr[`#interp.pointer<0> : `, Self.pointer_type]
-        }
+        self.value = __mlir_attr[`#interp.pointer<0> : `, Self.pointer_type]
```

Comment:
This should be `self.pointer_type` instead of `Self.pointer_type`?

Reply:
Why is that so? `Self.pointer_type` already carries `T`.